content/actions/learn-github-actions/variables.md

```diff
@@ -76,7 +76,7 @@ jobs:
 
 {% endraw %}
 
-You can access `env` variable values using runner environment variables or using contexts. The example above shows three custom variables being used as environment variables in an `echo` command: `$DAY_OF_WEEK`, `$Greeting`, and  `$First_Name`. The values for these variables are set, and scoped, at the workflow, job, and step level respectively. For more information on accessing variable values using contexts, see "[Using contexts to access variable values](#using-contexts-to-access-variable-values)."
+You can access `env` variable values using runner environment variables or using contexts. The example above shows three custom variables being used as runner environment variables in an `echo` command: `$DAY_OF_WEEK`, `$Greeting`, and `$First_Name`. The values for these variables are set, and scoped, at the workflow, job, and step level respectively. The interpolation of these variables happens on the runner, i.e. the `run` step. All other parts of a workflow are interpolated by {% data variables.product.prodname_actions %} and are not sent to the runner. In that case you must use a context to access the variable value. For more information on accessing variable values using contexts, see "[Using contexts to access variable values](#using-contexts-to-access-variable-values)."
```

Comment:
What does "referenced actions" refer to here?

Reply:
This means any actions you reference in a workflow, like:
```
steps:
  - uses: actions/checkout@v2
  - name: Set up Ruby
    uses: ruby/setup-ruby@55283cc23133118229fd3f97f9336ee23a179fcf 
```
where we reference the `actions/checkout` action and the `ruby/setup-ruby` action.

These actions will have `run` steps of their own, which may contain variables.

Reply:
Thanks for this clarification! I had a feeling that's what it meant. 

The confusion it brought back to me was in the scenario where a referenced action has `with:` options, such as: 
```
steps:
  - uses: actions/checkout@v2
  - name: Set up Ruby
  - uses: ruby/setup-ruby
     with: 
       ruby-version: '3.2'
``` 

Part of my initial confusion with this document was that I believed the `ruby-version: '3.2'` could use the Linux bash environment variable syntax instead of contexts.

What do you think about updating the text to read something like this? 
```
Other than `run` steps in workflows you write and `run` steps in referenced actions, ...
```

Reply:
Thanks. I've rewritten the text a little to take this into account. I'm going to go ahead and commit this so that we can get this merged.

Many thanks for your work on this.